main/main.go

```diff
@@ -159,7 +159,11 @@ func main() {
 	}
 
 	// Create relayers for each of the subnets configured as a source
+	// Each relayer goroutine should persist for the lifetime of the program,
+	// so we exit if any of them fail. In the case of a consistently failing
+	// goroutine, that blockchain should likely be disabled in the configuration.
 	var wg sync.WaitGroup
+	wg.Add(1) // Single counter to exit as soon as any relayer goroutine fails
```

Comment:
Do we just want to get rid of the health check then? It no longer does anything if we exit on the first goroutine failure.

Reply:
Still thinking it through, but I think we should probably keep the original health check. We could panic in certain situations, such as if there's no db entry for a chain and no start height set, which gets us the immediate exit in that scenario but not all

Reply:
Wait... that's already implemented lol. I'll just revert the `wg` change altogether

Reply:
IMO we should only panic if we encounter something that should otherwise be impossible. Exiting the application should only be done from the top level `main`, or via the health check. Otherwise error handling can become quickly intractible

Reply:
I agree with those points for something like `ASSERT` or `os.exit()` in c/c++, but panic in golang is a fair bit different. Idiomatically, panics should be thrown for "should be impossible" situations, but also errors that we know are unrecoverable such as a failed db, misconfiguration, etc. 

IMO, panics actually make error-handling easier, as it will do all the things that bubbling up an error would do - unwinds the stack, executing `defer`'d functions, but also prints a stack-trace which our current error handling doesn't do. It also makes it clear in the code at the earliest possible point that *if this errors, nothing above this can fix it*. The logger we use also contains functions like `StopOnPanic` to catch the panic, log the error, then continue panicing. Repos like `subnet-evm` use panic quite liberally.

All this being said, the failure recovery PR I almost have ready, uses a context instead to cancel the other goroutines, which probably works slightly better because it allows the other goroutines to unwind gracefully.